app/models/popular_article.rb

```diff
@@ -0,0 +1,19 @@
+class PopularArticle < ActiveRecord::Base
+
+  has_and_belongs_to_many :articles, through: :articles
+
+  validate :articles, :validate_popular_article_max
+  validate :articles, :validate_popular_article_min
+
+  def validate_popular_article_max
+    if self.articles.size > 3
```

Comment:
I think you might be able to get away with `validates :articles, length: { is: 3 }` instead of this custom stuff.

If not, you should remove `self` and add errors on `:article` instead.